homeassistant/components/sensor/filter.py

```diff
@@ -195,6 +225,31 @@ def device_state_attributes(self):
         return state_attr
 
 
+class FilterState(object):
+    """State abstraction for filter usage."""
+
+    def __init__(self, state):
+        """Initialize with HA State object."""
+        self.timestamp = state.last_updated
+        try:
+            self.state = float(state.state)
+        except ValueError:
```

Comment:
This is dangerous. This means that if a non float state comes in, all filters will blow up because they are still assuming it's a float?

Reply:
We already had this discussion :)

Some filters like Throttle can work with non Numeric states.

I simply moved the conversion to float from each of the numeric filters into the parent class filter.